core/Plugin/Menu.php

```diff
@@ -194,11 +197,40 @@ public function urlForDefaultUserParams($websiteId = false, $defaultPeriod = fal
         if (empty($websiteId)) {
             throw new \Exception("A website ID was not specified and a website to default to could not be found.");
         }
+        if (empty($defaultPeriod)) {
+            $defaultPeriod = $userPreferences->getDefaultPeriod(false);
+        }
         if (empty($defaultDate)) {
             $defaultDate = $userPreferences->getDefaultDate();
         }
-        if (empty($defaultPeriod)) {
-            $defaultPeriod = $userPreferences->getDefaultPeriod(false);
+
+        if ($defaultPeriod !== 'range' && !empty($defaultDate) && $defaultDate !== 'today') {
+            // not easy to make it work for range... is rarely the default anyway especially when just setting up
+            // Matomo as this logic is basically only applied on the first day a site is created
+            // no need to run logic when today is selected. It basically runs currently only when "yesterday" is selected
+            // as a default date but would also support future new default dates like past month etc.
+            try {
+                $siteCreationDate = Site::getCreationDateFor($websiteId);
+                $siteTimezone = Site::getTimezoneFor($websiteId);
+
+                if (!empty($siteCreationDate)) {
+                    if (is_numeric($defaultDate)) {
+                        $defaultDate = (int) $defaultDate; //prevent possible exception should defaultDate be a string timestamp
+                    }
+                    $siteCreationDate = Date::factory($siteCreationDate, $siteTimezone);
+                    $defaultDateObj = Date::factory($defaultDate, $siteTimezone);
+
+                    $period = Period\Factory::build($defaultPeriod, $defaultDateObj);
+                    $endDate = $period->getDateEnd();
+
+                    if (!$endDate->isLater($siteCreationDate)) {
+                        // when selected date is before site creation date or it is the site creation day
+                        $defaultDate = $siteCreationDate->toString();
+                    }
```

Comment:
This sets the date to load to the site creation date if the default period end date is past the site creation date correct? So if the site creation date is 2015-02-02, and `$defaultDate` is 2020-01-01 w/ period = year, then the date is always set to 2015-02-02. Is this the intended functionality? It doesn't sound like what the PR title implies.

Reply:
👍 adjusted the PR title @diosmosis

Reply:
But then should it be `$endDate->isEarlier($siteCreationDate)`? To set it "if select date is before site creation date"?

Reply:
it's pretty much the same or not? `!$endDate->isLater == $endDate->isEarlier()`? Will change the code ... tests still pass that way

Reply:
Oh I see, I missed the `!` while reading, LGTM